pkg/driver/driver_openstack.go

```diff
@@ -143,8 +146,50 @@ func (d *OpenStackDriver) Create() (string, string, error) {
 	var podNetworkIds = make(map[string]struct{})
 
 	if len(networkID) > 0 {
-		serverNetworks = append(serverNetworks, servers.Network{UUID: networkID})
+		// network port dealing if subnetID is specified
+
+		klog.V(3).Infof("existing network %s specified, need to pre-allocate ports ", networkID)
+
+		if subnetID != nil && len(*subnetID) > 0 {
+			// create port in given subnet
+			_, err := subnets.Get(nwClient, *subnetID).Extract()
+			if err != nil {
+				metrics.APIFailedRequestCount.With(prometheus.Labels{"provider": "openstack", "service": "neutron"}).Inc()
+				return "", "", fmt.Errorf("failed to get subnet information for subnetID %s: %s", *subnetID, err)
```

Comment:
```go
				return "", "", fmt.Errorf("failed to get subnet information for subnetID %q: %s", *subnetID, err)
```
Personal preference but when it comes to names `%q` is a bit better. This just adds quotes around the string.

Reply:
Ditto for other places

Reply:
Used %q for names but left %s for uids.